File: spring/new-hire-onboard-self-help-portal/src/main/java/com/example/newhireonboardselfhelpportal/entities/TeamChecklist.java
package com.example.newhireonboardselfhelpportal.entities;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

/**
 * TeamChecklist
 */
@Entity(name = "team_checklists")
public class TeamChecklist {
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Long id;

    @Column(name = "checklist_id")
    private Long checklistId;

    @Column(name = "team_id")
    private Long teamId;


    public Long getId() {
        return this.id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getChecklistId() {
        return this.checklistId;
    }

    public void setChecklistId(Long checklistId) {
        this.checklistId = checklistId;
    }

    public Long getTeamId() {
        return this.teamId;
    }

    public void setTeamId(Long teamId) {
        this.teamId = teamId;
    }

}

File: typeorm/new-hire-onboard-self-help-portal/src/entity/TeamChecklist.ts
import {Entity, PrimaryGeneratedColumn, ManyToOne, JoinColumn} from "typeorm";
import { Checklist } from "./Checklist";
import { Team } from "./Team";

@Entity({name: "team_checklists"})
export class TeamChecklist {

    @PrimaryGeneratedColumn()
    id: number;

    @ManyToOne(type => Checklist, checklist => checklist.teamChecklists)
    @JoinColumn({name: "checklist_id"})
    checklist: Checklist

    @ManyToOne(type => Team, team => team.teamChecklists)
    @JoinColumn({name: "team_id"})
    team: Team

}